### PR TITLE
Create new CColRefSet when merge EquivClass

### PR DESCRIPTION
**Pull request by terry-chelsea**
when merge EquivClass, it will modify child EquivClass if include new column in-place. such as a case below:

create table t1(id int not null, name text);

explain with t1 as (select id, count(1) as c from t1 group by id) select *, id from t1 union all select *, -1 as id from t1;
modified EquivClass will leading to adding invalid(error) LogicalSelect node when process PexprAddEqualityPreds

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@terry-chelsea Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=greenplum-db/gporca&pullRequestId=594)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=greenplum-db/gporca&pullRequestId=594) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@terry-chelsea Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=greenplum-db/gporca&pullRequestId=594)!

**Comment by zellerh** [comment]
This fix should also go into the other ORCA branches, [GPDB 6](https://github.com/greenplum-db/gpdb/tree/6X_STABLE) and [GPDB master](https://github.com/greenplum-db/gpdb). Would you like to submit a pull request for those as well or would you like me to handle them?

File: libgpopt/src/base/CUtils.cpp [hunk 1]
```diff
@@ -2772,9 +2772,10 @@ CUtils::PdrgpcrsMergeEquivClasses
 	for (ULONG ul = 0; ul < length; ul++)
 	{
 		CColRefSet *pcrs = (*pdrgpcrsSnd)[ul];
-		pcrs->AddRef();
+		// create new ColRefSet when include new columns. do not change original CColRefSet.
+		CColRefSet *pcrsCopyed = GPOS_NEW(mp) CColRefSet(mp, *pcrs);
 
-		CColRefSetArray *pdrgpcrs = PdrgpcrsAddEquivClass(mp, pcrs, pdrgpcrsMerged);
+		CColRefSetArray *pdrgpcrs = PdrgpcrsAddEquivClass(mp, pcrsCopyed, pdrgpcrsMerged);
```
Comment (chrishajas) [inline libgpopt/src/base/CUtils.cpp:2778]: nit: pcrsCopyed-> pcrsCopied

**Comment by chrishajas** [comment]
@terry-chelsea This has now been merged, thanks again for the PR and repro!

**Comment by terry-chelsea** [comment]
sorry for Late reply, I saw it had been merged into master, should i submit another pr to other branch?

**Comment by chrishajas** [comment]
@terry-chelsea We forward-merged it to gpdb master and 6X in the gpdb repo in https://github.com/greenplum-db/gpdb/commit/3062b121c7cf8decaf4f74a15438cb7e54cbaecb and https://github.com/greenplum-db/gpdb/commit/046f970a3785d3f3068efb104ca02514b2cd7019.

**Pull request merged**